components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -567,7 +568,142 @@ func generateVirtualService(instance *v1beta1.Notebook) (*unstructured.Unstructu
 	}
 
 	return vsvc, nil
+}
+
+func generateHTTPRoute(instance *v1beta1.Notebook) (*unstructured.Unstructured, error) {
+	name := instance.Name
```

Comment:
Did you consider using the gateway api package instead? https://pkg.go.dev/sigs.k8s.io/gateway-api It could help avoid setting ApiVersion, Kind, etc by defining an HTTPRoute object https://pkg.go.dev/sigs.k8s.io/gateway-api@v1.3.0/apis/v1#HTTPRoute

Reply:
(resolved)